### PR TITLE
DRAFT Compiler FE: Enable ReverseV2 operator 

### PR DESCRIPTION
**Pull request by d-krylov**
Enable ReverseV2 operator 

ONE-DCO-1.0-Signed-off-by: Denis Krylov <krylovdenism@gmail.com>

### CODE REVIEW

File: compiler/luci/export/src/CircleOperationExporter.cpp [hunk 1]
```diff
@@ -1280,6 +1281,19 @@ void OperationExporter::visit(luci::CircleReverseSequence *node)
...
 }
 
+void OperationExporter::visit(luci::CircleReverseV2 *node)
+{
+  uint32_t op_idx = md.registerBuiltinOpcode(circle::BuiltinOperator_REVERSE_V2);
+  std::vector<int32_t> inputs_vec{get_tensor_index(node->tensor()), get_tensor_index(node->axis())};
+  std::vector<int32_t> outputs_vec{get_tensor_index(static_cast<loco::Node *>(node))};
+  auto inputs = builder.CreateVector(inputs_vec);
+  auto outputs = builder.CreateVector(outputs_vec);
+  auto options = CreateReverseSequenceOptions(builder);
```
Comment (seanshpark) [inline compiler/luci/export/src/CircleOperationExporter.cpp:1291]: `ReverseSequence` and `ReverseV2` are different operators. Please use `ReverseV2Options`

File: compiler/luci/lang/include/luci/IR/Nodes/CircleReverseV2.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+namespace luci
+{
+
+/**
+ * @brief ReverseV2 in Circle
+ */
+class CircleReverseV2 final : public FixedArityNode<2, CircleNodeImpl<CircleOpcode::REVERSE_V2>>
+{
+public:
+  CircleReverseV2() = default;
```
Comment (seanshpark) [inline compiler/luci/lang/include/luci/IR/Nodes/CircleReverseV2.h:34]: Can you please remove this like the other IRs?

File: compiler/luci/service/src/CircleShapeInferenceRule.cpp [hunk 3]
```diff
@@ -1478,6 +1478,16 @@ class ShapeInferenceAlgorithm final : public luci::CircleNodeVisitor<loco::NodeS
     return loco::NodeShape{input_shape};
   }
 
+  loco::NodeShape visit(const luci::CircleReverseV2 *node) final
+  {
+    auto input_shape = loco::shape_get(node->tensor()).as<loco::TensorShape>();
+    auto axis_shape = loco::shape_get(node->axis()).as<loco::TensorShape>();
```
Comment (seanshpark) [inline compiler/luci/service/src/CircleShapeInferenceRule.cpp:1484]: `axis_shape `  is only used for `assert`, which will fail in release build.

Can you check this is working?
```cxx
LUCI_ASSERT(loco::shape_get(node->axis()).as<loco::TensorShape>().rank() == 1, "Tensor must be 1-D");
```

**Comment by seanshpark** [comment]
Others looks OK

**Comment by binarman** [comment]
everything is merged, thank you!

**Pull request closed without merging**